src/kprofile.cpp

```diff
@@ -19,15 +25,43 @@ KProfile::KProfile(QObject *parent) :
 }
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-void KProfile::startDownload(QWebEngineDownloadItem* download)
+void KProfile::openFile(QWebEngineDownloadItem* download)
 #else
-void KProfile::startDownload(QWebEngineDownloadRequest* download)
+void KProfile::openFile(QWebEngineDownloadRequest* download)
 #endif
 {
     QString defaultFileName = download->url().fileName();
-    QString fileName = QFileDialog::getSaveFileName(KiwixApp::instance()->getMainWindow(),
-                                                       gt("save-file-as-window-title"), defaultFileName);
+    QTemporaryFile *tempFile = new QTemporaryFile(QDir::tempPath() + "/XXXXXX." + QFileInfo(defaultFileName).suffix());
+    if (tempFile->open()) {
+        QString tempFilePath = tempFile->fileName();
+        tempFile->close();
+#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
+        download->setPath(tempFilePath);
+#else
+        download->setDownloadFileName(tempFilePath);
+#endif
+        connect(download, &QWebEngineDownloadItem::finished, [tempFilePath]() {
+            if(!QDesktopServices::openUrl(QUrl::fromLocalFile(tempFilePath)))
+                showInfoBox(gt("error-title"), gt("error-opening-file-with"), KiwixApp::instance()->getMainWindow());
+        });
+        download->accept();
+    } else {
```

Comment:
Memory leak - `tempFile` is not `delete`d. You can  create the `QTemporaryFile` object on the stack (rather than on the heap) and [`disable auto-remove`](https://doc.qt.io/qt-5/qtemporaryfile.html#setAutoRemove).

Reply:
Its on the stack now but I havent disabled auto-remove, why do we need that?

Reply:
Then the file will be deleted when `tempFile` goes out of scope. Since the file is opened asynchronously (after its download is finished) that operation may fail.